sdl-core/src/main/scala/io/smartdatalake/workflow/action/ProxyAction.scala

```diff
@@ -0,0 +1,98 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.smartdatalake.workflow.action
+
+import io.smartdatalake.communication.agent.AgentClient
+import io.smartdatalake.config.SdlConfigObject.DataObjectId
+import io.smartdatalake.config.{FromConfigFactory, SdlConfigObject}
+import io.smartdatalake.definitions.{Condition, ExecutionMode}
+import io.smartdatalake.util.dag.DAGHelper.NodeId
+import io.smartdatalake.util.spark.DataFrameUtil
+import io.smartdatalake.workflow.ExecutionPhase.ExecutionPhase
+import io.smartdatalake.workflow.agent.Agent
+import io.smartdatalake.workflow.dataframe.spark.{SparkDataFrame, SparkSubFeed}
+import io.smartdatalake.workflow.dataobject.DataObject
+import io.smartdatalake.workflow.{ActionPipelineContext, ExecutionPhase, InitSubFeed, SubFeed}
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.types.StructType
+
+case class ProxyAction(wrappedAction: Action, override val id: SdlConfigObject.ActionId, agent: Agent) extends Action {
+
+  override def factory: FromConfigFactory[Action] = wrappedAction.factory
+
+  override def nodeId: NodeId = wrappedAction.nodeId
+
+  override def atlasName: String = wrappedAction.atlasName
+
+  override def metadata: Option[ActionMetadata] = wrappedAction.metadata
+
+  override def inputs: Seq[DataObject] = wrappedAction.inputs
+
+  override def outputs: Seq[DataObject] = wrappedAction.outputs
+
+  override def executionCondition: Option[Condition] = wrappedAction.executionCondition
+
+  override def executionMode: Option[ExecutionMode] = wrappedAction.executionMode
+
+  override def metricsFailCondition: Option[String] = wrappedAction.metricsFailCondition
+
+   override def prepare(implicit context: ActionPipelineContext): Unit = {
+     runOnAgent( ExecutionPhase.Prepare)
+  }
+
+  override def init(subFeeds: Seq[SubFeed])(implicit context: ActionPipelineContext): Seq[SubFeed] = {
+    runOnAgent( ExecutionPhase.Init)
+  }
+
+  override def exec(subFeeds: Seq[SubFeed])(implicit context: ActionPipelineContext): Seq[SubFeed] = {
+    runOnAgent(ExecutionPhase.Exec)
+  }
+
+  def runOnAgent(executionPhase: ExecutionPhase)(implicit context: ActionPipelineContext): Seq[SubFeed] = {
+    val agentClient = AgentClient(agent)
+    val hoconInstructions = AgentClient.prepareHoconInstructions(wrappedAction, context.instanceRegistry.getConnections, agent, executionPhase)
+    agentClient.sendSDLMessage(hoconInstructions)
+
+    val instructionId = hoconInstructions.agentInstruction.get.instructionId
+
+    while (agentClient.socket.isConnected && !agentClient.socket.pendingResults.contains(instructionId)) {
+      Thread.sleep(1000)
+      println(s"Waiting for ${agent.id.id} to finish $instructionId...")
+    }
+    if(!agentClient.socket.isConnected){
```

Comment:
We should check what happens when the connection is temporarily lost. If you have long-running actions, it would not be so nice if the job crashes because you had a network instability over this time.

Reply:
We should discuss how far we want to implement the communication capabilities with Jetty. For me it's just the default implementation of the AgentClient, but in practice I imagine that it will rarely be used and instead we will mostly use AzureRelayAgentClient or their alternatives for Google or AWS

Reply:
Yes, we should discuss this, because in order to use Jetty in production we will probably need to set up authentication and encryption. Regarding the original comment: Azure Relay can deal with temporarily lost connections as long as there is an active connection when the messages are sent.

Reply:
"as long as there is an active connection when the messages are sent." What does this mean in practice for us?

Reply:
It means it is fine if you have some long running job on the AgentServer and you loose connection as long as you regain the connection before sending the response. I think this is enough for us because the probability that your job crashes because there is a short connection loss is small. If there is a longer network problem there is not much you can do anyway.